Source/WTF/wtf/FixedBitVector.h

```diff
@@ -0,0 +1,300 @@
+
+    unsigned hash() const;
+
+    void dump(PrintStream& out) const;
+
+    class iterator {
+        WTF_MAKE_FAST_ALLOCATED;
+
+    public:
+        iterator()
+            : m_bitVector(nullptr)
+            , m_index(0)
+        {
+        }
+
+        iterator(const FixedBitVector& bitVector, size_t index)
+            : m_bitVector(&bitVector)
+            , m_index(index)
+        {
+        }
+
+        size_t operator*() const { return m_index; }
+
+        iterator& operator++()
+        {
+            m_index = m_bitVector->findBit(m_index + 1, true);
+            return *this;
+        }
+
+        iterator operator++(int)
+        {
+            iterator result = *this;
+            ++(*this);
+            return result;
+        }
+
+        bool isAtEnd() const
+        {
+            return m_index >= m_bitVector->size();
+        }
+
+        bool operator==(const iterator& other) const
+        {
+            return m_index == other.m_index;
+        }
+
+        bool operator!=(const iterator& other) const
+        {
+            return !(*this == other);
+        }
+
+    private:
+        const FixedBitVector* m_bitVector;
+        size_t m_index;
+    };
+
+    // Use this to iterate over set bits.
+    iterator begin() const
+    {
+        ASSERT(m_bits);
+        return iterator(*this, findBit(0, true));
+    }
+
+    iterator end() const
+    {
+        ASSERT(m_bits);
+        return iterator(*this, size());
+    }
+
+private:
+    bool get(size_t bitIndex, Dependency = Dependency()) const;
+
+    static constexpr unsigned wordSize = sizeof(WordType) * 8;
+    static constexpr WordType one = 1;
+
+    size_t m_size { 0 };
+    unsigned m_words { 0 };
+    WordType* m_bits { nullptr };
+};
+
+ALWAYS_INLINE bool FixedBitVector::concurrentAllocateOnce(size_t size)
+{
+    return bitwise_cast<Atomic<WordType*>*>(&m_bits)->transactionRelaxed(
+        [&](WordType*& value) -> bool {
+            if (value)
+                return false;
+
+            m_size = size;
+            m_words = (size + wordSize - 1) / wordSize;
+            value = (WordType*)FixedBitVectorMalloc::tryZeroedMalloc(m_words * sizeof(WordType));
+            return true;
+        });
+}
+
+ALWAYS_INLINE void FixedBitVector::allocateOnce(size_t size)
+{
+    if (m_bits)
+        return;
+
+    m_size = size;
+    m_words = (size + wordSize - 1) / wordSize;
+    m_bits = (WordType*)FixedBitVectorMalloc::tryZeroedMalloc(m_words * sizeof(WordType));
+}
+
+ALWAYS_INLINE bool FixedBitVector::concurrentTest(size_t bitIndex, Dependency dependency)
+{
+    if (!m_bits || bitIndex >= m_size)
+        return false;
+
+    WordType mask = one << (bitIndex % wordSize);
+    size_t wordIndex = bitIndex / wordSize;
+    WordType* data = dependency.consume(m_bits) + wordIndex;
+    return bitwise_cast<Atomic<WordType>*>(data)->transactionRelaxed(
+        [&](WordType& value) -> bool {
+            return value & mask;
+        });
+}
```

Comment:
It doesn't make sense to have a `concurrentTest`.  This does not fix and race in the client.  An we should not have any race internally because we should never be reallocating the backing store.  Remove this.